### ISSUE TITLE
Does not detect anonymous function expressions

### ISSUE THREAD

**Issue by silverwind** [SebastienGllmt/eslint-plugin-no-floating-promise#8]
This works and is detected:

```js
async function a() {};
a();
```

But this is not detected (and passes):

```js
const a = async () => {};
a();
```

**Comment by papb** [comment]
Good catch, I think your second example should be caught as well. But if you use `let` instead of `const`, then I think it should not be caught.
